packages/core/src/feature-service-registry.ts

```diff
@@ -250,14 +266,14 @@ export class FeatureServiceRegistry implements FeatureServiceRegistryLike {
 
     const bindings = new Map<string, FeatureServiceBinding<unknown>>();
     const featureServices: FeatureServices = Object.create(null);
-    const allDependencies = {...optionalDependencies, ...dependencies};
+    const allDependencies = mergeFeatureServiceDependencies(consumerDefinition);
 
     for (const providerId of Object.keys(allDependencies)) {
       const binding = this.bindFeatureService(
         providerId,
         consumerUid,
         allDependencies[providerId],
-        {optional: !dependencies.hasOwnProperty(providerId)}
+        {optional: !dependencies.featureServices.hasOwnProperty(providerId)}
```

Comment:
extract to helper function `isOptionalFeatureServiceDependency` and remove `dependencies` destructuring